src/AspNetCore/Src/forwardinghandler.cxx

```diff
@@ -1534,7 +1532,38 @@ REQUEST_NOTIFICATION_STATUS
             reinterpret_cast<PVOID>(static_cast<DWORD_PTR>(hrCompletionStatus)));
     }
 
-    if (m_pApplication->QueryConfig()->QueryHostingModel() == HOSTING_OUT_PROCESS)
+    if (m_pApplication->QueryConfig()->QueryHostingModel() == HOSTING_IN_PROCESS)
+    {
+        if (fPostNotification)
+        {
+            return RQ_NOTIFICATION_FINISH_REQUEST;
+        }
+
+        if (FAILED(hrCompletionStatus))
```

Comment:
We shouldn't do this here. What if there's a pending async operation in managed code? It'll never complete.

Reply:
I think we need to start checking the number of bytes written. We need to also confirm that the result of the read/write does not return the "compressed" or wrong value.